src/fastcs/transport/epics/p4p/handlers.py

```diff
@@ -0,0 +1,119 @@
+import asyncio
+import time
+from collections.abc import Callable
+
+import numpy as np
+from p4p.nt import NTScalar
+from p4p.server import ServerOperation
+from p4p.server.asyncio import SharedPV
+
+from fastcs.attributes import Attribute, AttrR, AttrRW, AttrW
+from fastcs.datatypes import Table
+
+from .types import (
+    MAJOR_ALARM_SEVERITY,
+    RECORD_ALARM_STATUS,
+    cast_from_p4p_value,
+    cast_to_p4p_value,
+    get_p4p_type,
+    p4p_alarm_states,
+)
+
+
+class AttrWHandler:
+    def __init__(self, attr_w: AttrW | AttrRW):
+        self._attr_w = attr_w
+
+    async def put(self, pv: SharedPV, op: ServerOperation):
+        value = op.value()
+        if isinstance(value, list):
+            assert isinstance(self._attr_w.datatype, Table)
```

Comment:
I'd invert these in case we're sent a list for another reason:
```suggestion
        if isinstance(self._attr_w.datatype, Table):
            assert isinstance(value, list)
```